In [2]:
import pandas as pd
import plotly.express as px
import numpy as np

In [3]:
source_df = pd.read_csv("../data/processed/preprocessed_data.csv")
preprocessed_df = pd.read_csv("../data/processed/preprocessed_data.csv")

preprocessed_df = preprocessed_df[preprocessed_df["Искомая позиция"] != "MLOps Engineer"]

source_df.shape, preprocessed_df.shape

((18370, 13), (15176, 13))

In [4]:
preprocessed_df.head()

Дата обновления резюме  Возраст        ЗП            Город  \
0             2024-12-05       40       NaN           Тамбов   
1             2024-12-05       35  200000.0           Москва   
2             2024-12-05       41  300000.0  Санкт-Петербург   
3             2024-12-05       56       NaN           Москва   
4             2024-12-05       30   90000.0           Москва   

                                      Условия работы         Занятость  \
0            удаленная работа, готов к командировкам  полная занятость   
1                              готов к командировкам  полная занятость   
2            готов к переезду, готов к командировкам  полная занятость   
3                           не готов к командировкам  полная занятость   
4  готов к переезду: Кемерово, не готов к команди...  полная занятость   

                                              Навыки  \
0  .NET Core, .NET Framework, Git, PostgreSQL, Bi...   
1                                                NaN   
2                                                NaN   
3                                                NaN   
4                                                 C#   

                      Последнее/текущее место работы  \
0                                     Hybrid, Тамбов   
1  АО "БОЛЬШЕВИЧКА" (швейная фабрика и сеть из 14...   
2                       ООО "Пекод", Санкт-Петербург   
3                          ИП Азнаурян И. Э., Москва   
4                           CORALINA DIGITAL, Москва   

                         Последняя/текущая должность  \
0                        C# (.NET) Backend Developer   
1                                Инженер-программист   
2                   Web-программист, Web-разработчик   
3  Программист-разработчик (PHP, Smarty (PHP), Ja...   
4                                     Разработчик C#   

                                   Образование и ВУЗ  \
0  Тамбовский государственный технический универс...   
1  Белорусский государственный университет информ...   
2  Санкт-Петербургский политехнический университе...   
3  Московский авиационный институт (Национальный ...   
4  Московский государственный технический универс...   

                                    Ссылка на резюме Искомая позиция  \
0    https://www.superjob.ru//resume/c-55328505.html    C# Developer   
1  https://www.superjob.ru//resume/programmist-53...    C# Developer   
2  https://www.superjob.ru//resume/programmist-c-...    C# Developer   
3  https://www.superjob.ru//resume/web-programmis...    C# Developer   
4  https://www.superjob.ru//resume/razrabotchik-c...    C# Developer   

  pipeline_load_date  
0         2024-12-05  
1         2024-12-05  
2         2024-12-05  
3         2024-12-05  
4         2024-12-05

## Количество соискателей по каждой позиции

In [5]:
category_counts = preprocessed_df["Искомая позиция"].value_counts().reset_index()

category_counts.columns = ["Искомая позиция", "Количество соискателей"]

category_counts

Искомая позиция  Количество соискателей
0            Golang Developer                    3164
1              Java Developer                    2768
2            Python Developer                    2748
3              Rust Developer                    2725
4          Frontend Developer                    1763
5                C# Developer                     840
6             DevOps Engineer                     536
7                Data Analyst                     339
8               Data Engineer                     243
9              Data Scientist                      46
10  Machine Learning Engineer                       4

In [6]:
fig = px.bar(category_counts,
             x="Количество соискателей", 
             y="Искомая позиция", 
             title="Количество соискателей по каждой позиции по всей России",
             labels={"ЗП": "Заработная плата", "Искомая позиция": "Искомая позиция"},
             text="Количество соискателей",
             height=700,
             width=1200,
             color="Количество соискателей",
             color_continuous_scale="rainbow"
             )

fig.update_layout(
    xaxis_title="Искомая позиция",
    yaxis_title="Количество соискателей",
    title_x=0.5 ,
)

fig.update_traces(textposition="outside",
                  textfont=dict(
                      size=15,
                      color="black",
                      family="Arial"
                      ))

fig.update_traces(marker=dict(line=dict(width=0.5, color='black')))

fig.show()

## Сколько человек скрывают желаемую зарплату

In [7]:
num_salary_available = len(preprocessed_df[~preprocessed_df["ЗП"].isna()])
num_salary_unavailable = len(preprocessed_df[preprocessed_df["ЗП"].isna()])
num_salary_available, num_salary_unavailable

(6455, 8721)

In [8]:
salary_num_df = pd.DataFrame({"ЗП": ["Открыта", "Скрыта"],
                              "Количество": [num_salary_available, num_salary_unavailable]})

salary_num_df

ЗП  Количество
0  Открыта        6455
1   Скрыта        8721

In [9]:
fig = px.bar(salary_num_df,
             x="ЗП",
             y="Количество",
             text_auto="Количество",
             orientation="v",
             height=700,
             color='ЗП')

fig.update_layout(
    xaxis_title="Зарплата",
    yaxis_title="Количество",
    title_x=0.5 , # Выравнивание заголовка по центру
)

fig.update_traces(text=salary_num_df["Количество"],
                  textposition="outside",
                                    textfont=dict(
                      size=18,
                      color="black",
                      family="Arial"
                      ))

fig.update_traces(marker=dict(line=dict(width=0.5, color='black')))

fig.show()

## Распределение желаемой зарплаты по специализациям

In [10]:
median_salary_by_positions = preprocessed_df.groupby('Искомая позиция')['ЗП'].median().reset_index()
median_salary_by_positions.rename(columns={'ЗП': 'Медиана'}, inplace=True)
median_salary_by_positions

Искомая позиция   Медиана
0                C# Developer   80000.0
1                Data Analyst   90000.0
2               Data Engineer  100000.0
3              Data Scientist   80000.0
4             DevOps Engineer  170000.0
5          Frontend Developer   80000.0
6            Golang Developer   80000.0
7              Java Developer   80000.0
8   Machine Learning Engineer       NaN
9            Python Developer   70000.0
10             Rust Developer   90000.0

In [11]:
fig = px.box(preprocessed_df[["Искомая позиция", "ЗП"]], 
             x="ЗП", 
             y="Искомая позиция", 
             title="Boxplots ЗП по искомой позиции по всей России",
             labels={"ЗП": "Заработная плата", "Искомая позиция": "Искомая позиция"},
             height=700,
             color="Искомая позиция"
             )

fig.update_layout(
    xaxis=dict(
        tickformat='.2s',  # Использование разделителей тысяч
        title="Заработная плата",  # Заголовок оси Y
    ),
    yaxis_title="Искомая позиция",
    title_x=0.5,
    showlegend=False
)

for position in median_salary_by_positions['Искомая позиция']:
    median_value = median_salary_by_positions.loc[median_salary_by_positions['Искомая позиция'] == position, 'Медиана'].values[0]
    fig.add_annotation(
        x=median_value,
        y=position,
        text=f"{median_value:.0f}",
        showarrow=False,
        font=dict(color="black", size=12),
        opacity=0.6,
        align="center",
        # xshift=10,
    )

fig.show()

## Распределение зарплат по специализациям в Москве

In [12]:
median_salary_moscow_by_positions = preprocessed_df[preprocessed_df["Город"] == "Москва"].groupby('Искомая позиция')['ЗП'].median().reset_index()
median_salary_moscow_by_positions.rename(columns={'ЗП': 'Медиана'}, inplace=True)
median_salary_moscow_by_positions

Искомая позиция   Медиана
0                C# Developer   87500.0
1                Data Analyst   95000.0
2               Data Engineer  100000.0
3              Data Scientist   95000.0
4             DevOps Engineer  200000.0
5          Frontend Developer  100000.0
6            Golang Developer  100000.0
7              Java Developer  100000.0
8   Machine Learning Engineer       NaN
9            Python Developer   80000.0
10             Rust Developer  100000.0

In [13]:
fig = px.box(preprocessed_df[preprocessed_df["Город"] == "Москва"][["Искомая позиция", "ЗП"]], 
             x="ЗП", 
             y="Искомая позиция",
             title="Boxplots ЗП по искомой позиции по Москве",
             labels={"ЗП": "Заработная плата", "Искомая позиция": "Искомая позиция"},
             height=700,
             color="Искомая позиция")

fig.update_layout(
    xaxis=dict(
        tickformat='.2s',
        title="Заработная плата",
    ),
    yaxis_title="Искомая позиция",
    title_x=0.5,
    showlegend=False
)

for position in median_salary_moscow_by_positions['Искомая позиция']:
    median_value = median_salary_moscow_by_positions.loc[median_salary_by_positions['Искомая позиция'] == position, 'Медиана'].values[0]
    fig.add_annotation(
        x=median_value,
        y=position,
        text=f"{median_value:.0f}",
        showarrow=False,
        font=dict(color="black", size=12),
        opacity=0.6,
        align="center",
        # xshift=10,
    )

fig.show()

## Распределение зарплат по специализациям в Санкт-Петербурге

In [14]:
median_salary_spb_by_positions = preprocessed_df[preprocessed_df["Город"] == "Санкт-Петербург"].groupby('Искомая позиция')['ЗП'].median().reset_index()
median_salary_spb_by_positions.rename(columns={'ЗП': 'Медиана'}, inplace=True)
median_salary_spb_by_positions

Искомая позиция   Медиана
0        C# Developer  100000.0
1        Data Analyst  140000.0
2       Data Engineer  200000.0
3      Data Scientist   65000.0
4     DevOps Engineer  165000.0
5  Frontend Developer   70000.0
6    Golang Developer   80000.0
7      Java Developer   80000.0
8    Python Developer   70000.0
9      Rust Developer   90000.0

In [15]:
fig = px.box(preprocessed_df[preprocessed_df["Город"] == "Санкт-Петербург"][["Искомая позиция", "ЗП"]], 
             x="ЗП", 
             y="Искомая позиция", 
             title="Boxplots ЗП по искомой позиции по Санкт-Петербургу",
             labels={"ЗП": "Заработная плата", "Искомая позиция": "Искомая позиция"},
             height=700,
             color="Искомая позиция")

fig.update_layout(
    xaxis=dict(
        tickformat='.2s',  # Использование разделителей тысяч
        title="Заработная плата",  # Заголовок оси Y
    ),
    yaxis_title="Искомая позиция",
    title_x=0.5,
    showlegend=False
)

for position in median_salary_spb_by_positions['Искомая позиция']:
    median_value = median_salary_spb_by_positions.loc[median_salary_spb_by_positions['Искомая позиция'] == position, 'Медиана'].values[0]
    fig.add_annotation(
        x=median_value,
        y=position,
        text=f"{median_value:.0f}",
        showarrow=False,
        font=dict(color="black", size=12),
        opacity=0.6,
        align="center",
        # xshift=10,
    )

fig.show()

## Распределение количества позиций по городам

Возьмем для рассмотрения только те города, где общее число позиций превышает 50

In [16]:
city_counts = preprocessed_df['Город'].value_counts().reset_index()
city_counts.columns = ['Город', 'Количество позиций']

city_counts = city_counts[city_counts["Количество позиций"] > 50]

city_counts = city_counts.sort_values("Количество позиций")[::-1]
city_counts.shape[0]

37

In [17]:
fig = px.bar(city_counts,
             x='Количество позиций',
             y='Город',
             orientation='h',
             title="Количество позиций по городам",
             height=700,
             labels={"Количество позиций": "Количество позиций", "Город": "Город"},
             text='Количество позиций',
             color="Количество позиций",
             color_continuous_scale="rainbow",
             )

fig.update_traces(textposition='outside',
                  texttemplate='%{text}',
                  textfont=dict(
                      size=18,
                      color="black",
                      family="Arial",
                      ))


fig.show()

## Распределение ожидаемых зарплат по городам

Возьмем для рассмотрения только те города, где общее число позиций превышает 50

In [18]:
city_counts = preprocessed_df['Город'].value_counts().reset_index()
city_counts.columns = ['Город', 'Количество позиций']

cities_to_include = city_counts[city_counts["Количество позиций"] > 50]
cities_to_include = cities_to_include["Город"].unique()

filtered_cities = preprocessed_df[preprocessed_df["Город"].isin(cities_to_include)]
filtered_cities.head()


Дата обновления резюме  Возраст        ЗП            Город  \
1             2024-12-05       35  200000.0           Москва   
2             2024-12-05       41  300000.0  Санкт-Петербург   
3             2024-12-05       56       NaN           Москва   
4             2024-12-05       30   90000.0           Москва   
6             2024-12-05       42   70000.0  Санкт-Петербург   

                                      Условия работы         Занятость  \
1                              готов к командировкам  полная занятость   
2            готов к переезду, готов к командировкам  полная занятость   
3                           не готов к командировкам  полная занятость   
4  готов к переезду: Кемерово, не готов к команди...  полная занятость   
6  готов к переезду: Москва, Саратов  и ещё 1 гор...  полная занятость   

                                  Навыки  \
1                                    NaN   
2                                    NaN   
3                                    NaN   
4                                     C#   
6  Работа в команде, 1С программирование   

                      Последнее/текущее место работы  \
1  АО "БОЛЬШЕВИЧКА" (швейная фабрика и сеть из 14...   
2                       ООО "Пекод", Санкт-Петербург   
3                          ИП Азнаурян И. Э., Москва   
4                           CORALINA DIGITAL, Москва   
6                                    Санкт-Петербург   

                         Последняя/текущая должность  \
1                                Инженер-программист   
2                   Web-программист, Web-разработчик   
3  Программист-разработчик (PHP, Smarty (PHP), Ja...   
4                                     Разработчик C#   
6                                     Программист С#   

                                   Образование и ВУЗ  \
1  Белорусский государственный университет информ...   
2  Санкт-Петербургский политехнический университе...   
3  Московский авиационный институт (Национальный ...   
4  Московский государственный технический универс...   
6                                                NaN   

                                    Ссылка на резюме Искомая позиция  \
1  https://www.superjob.ru//resume/programmist-53...    C# Developer   
2  https://www.superjob.ru//resume/programmist-c-...    C# Developer   
3  https://www.superjob.ru//resume/web-programmis...    C# Developer   
4  https://www.superjob.ru//resume/razrabotchik-c...    C# Developer   
6  https://www.superjob.ru//resume/programmist-1s...    C# Developer   

  pipeline_load_date  
1         2024-12-05  
2         2024-12-05  
3         2024-12-05  
4         2024-12-05  
6         2024-12-05

In [19]:
median_salary_by_cities = filtered_cities.groupby('Город')['ЗП'].median().reset_index()
median_salary_by_cities.rename(columns={'ЗП': 'Медиана'}, inplace=True)
median_salary_by_cities.head()

Город   Медиана
0   Белгород  170000.0
1     Брянск   65000.0
2   Владимир   80000.0
3  Волгоград   50000.0
4    Воронеж  115000.0

In [20]:
fig = px.box(filtered_cities,
             x="ЗП", 
             y="Город", 
             title="Boxplots ЗП по городам",
             labels={"ЗП": "Заработная плата", "Город": "Город"},
             height=1200,
             color="Город",
             )

fig.update_layout(
    xaxis=dict(
        tickformat='.2s',
        title="Заработная плата",
    ),
    yaxis_title="Искомая позиция",
    title_x=0.5,
    showlegend=False
)

for position in median_salary_by_cities['Город']:
    median_value = median_salary_by_cities.loc[median_salary_by_cities['Город'] == position, 'Медиана'].values[0]
    fig.add_annotation(
        x=median_value,
        y=position,
        text=f"{median_value:.0f}",
        showarrow=False,
        font=dict(color="black", size=12),
        opacity=0.6,
        align="center",
        # xshift=10,
    )

fig.show()

## Самые популярные ключевые навыки в IT

Рассмотрим топ-30 навыков по частоте встречаемости в резюме соискателей

In [21]:
skills_split = preprocessed_df['Навыки'].dropna().str.split(', ').explode().str.strip()

skills_counts = skills_split.value_counts()

skills_filtered = skills_counts[:30].reset_index()

skills_filtered.columns = ["Навык", "Частота"]

skills_filtered.head()

Навык  Частота
0         Git     4446
1         SQL     4104
2      Python     3771
3  JavaScript     2597
4  PostgreSQL     2546

In [22]:
fig = px.bar(skills_filtered,
             x="Частота",
             y="Навык",
            #  orientation='h',
             labels={'x': 'Навык', 'y': 'Частота'},
             title="Частота встречаемости навыков у соискателей по всем позициям",
             text='Частота',
             height=700,
             color="Частота",
             color_continuous_scale="rainbow",
             )

fig.update_traces(textposition='outside',
                  texttemplate='%{text}',
                  textfont=dict(
                      size=18,
                      color="black",
                      family="Arial",
                      ))

fig.show()

## Самые популярные ключевые навыки по специализации Golang Developer

In [23]:
skills_split_golang = preprocessed_df[preprocessed_df['Искомая позиция'] == 'Golang Developer']['Навыки'].dropna().str.split(', ').explode().str.strip()

skills_counts_golang = skills_split_golang.value_counts()

skills_filtered_golang = skills_counts_golang[:30].reset_index()

skills_filtered_golang.columns = ["Навык", "Частота"]

skills_filtered_golang.head()

Навык  Частота
0     Git      957
1     SQL      952
2  Python      644
3    Java      634
4      C#      628

In [24]:
fig = px.bar(skills_filtered_golang,
             x="Частота",
             y="Навык",
            #  orientation='h',
             labels={'x': 'Навык', 'y': 'Частота'},
             title="Частота встречаемости навыков у соискателей по позиции Golang Developer",
             text='Частота',
             height=700,
             color="Частота",
             color_continuous_scale="rainbow",
             )

fig.update_traces(textposition='outside',
                  texttemplate='%{text}',
                  textfont=dict(
                      size=18,
                      color="black",
                      family="Arial",
                      ))

fig.show()

## Самые популярные ключевые навыки по специализации Frontend Developer

In [25]:
skills_split_frontend = preprocessed_df[preprocessed_df['Искомая позиция'] == 'Frontend Developer']['Навыки'].dropna().str.split(', ').explode().str.strip()

skills_counts_frontend = skills_split_frontend.value_counts()

skills_filtered_frontend = skills_counts_frontend[:30].reset_index()

skills_filtered_frontend.columns = ["Навык", "Частота"]

skills_filtered_frontend.head()

Навык  Частота
0  JavaScript      798
1         Git      587
2       React      497
3  TypeScript      440
4        HTML      405

In [26]:
fig = px.bar(skills_filtered_frontend,
             x="Частота",
             y="Навык",
            #  orientation='h',
             labels={'x': 'Навык', 'y': 'Частота'},
             title="Частота встречаемости навыков у соискателей по позиции Frontend Developer",
             text='Частота',
             height=700,
             color="Частота",
             color_continuous_scale="rainbow",
             )

fig.update_traces(textposition='outside',
                  texttemplate='%{text}',
                  textfont=dict(
                      size=18,
                      color="black",
                      family="Arial",
                      ))

fig.show()

## Самые популярные ключевые навыки по специализации DevOps Engineer


In [27]:
skills_split_devops = preprocessed_df[preprocessed_df['Искомая позиция'] == 'DevOps Engineer']['Навыки'].dropna().str.split(', ').explode().str.strip()

skills_counts_devops = skills_split_devops.value_counts()

skills_filtered_devops = skills_counts_devops[:30].reset_index()

skills_filtered_devops.columns = ["Навык", "Частота"]

skills_filtered_devops.head()

Навык  Частота
0         Git      120
1       Linux      114
2      Python       95
3         SQL       91
4  PostgreSQL       77

In [28]:
fig = px.bar(skills_filtered_devops,
             x="Частота",
             y="Навык",
            #  orientation='h',
             labels={'x': 'Навык', 'y': 'Частота'},
             title="Частота встречаемости навыков у соискателей по позиции DevOps Engineer",
             text='Частота',
             height=700,
             color="Частота",
             color_continuous_scale="rainbow",
             )

fig.update_traces(textposition='outside',
                  texttemplate='%{text}',
                  textfont=dict(
                      size=18,
                      color="black",
                      family="Arial",
                      ))

fig.show()

## Распределение возраста по специализациям

In [29]:
median_ages_by_position = preprocessed_df.groupby('Искомая позиция')['Возраст'].median().reset_index()
median_ages_by_position.rename(columns={'Возраст': 'Медиана'}, inplace=True)
median_ages_by_position

Искомая позиция  Медиана
0                C# Developer     24.0
1                Data Analyst     31.0
2               Data Engineer     33.0
3              Data Scientist     36.0
4             DevOps Engineer     32.0
5          Frontend Developer     27.0
6            Golang Developer     29.0
7              Java Developer     29.0
8   Machine Learning Engineer     29.0
9            Python Developer     24.0
10             Rust Developer     29.0

In [30]:
fig = px.box(preprocessed_df,
             x="Возраст",
             y="Искомая позиция",
             title='Возраст соискателей для каждой искомой позиции',
             labels={'Возраст': 'Возраст', 'Искомая позиция': 'Искомая позиция'},
             height=700,
             color="Искомая позиция",
             )

fig.update_layout(
    xaxis=dict(
        tickformat='.2s',
        title="Возраст",
    ),
    yaxis_title="Искомая позиция",
    title_x=0.5,
    showlegend=False
)

for position in median_ages_by_position['Искомая позиция']:
    median_value = median_ages_by_position.loc[median_ages_by_position['Искомая позиция'] == position, 'Медиана'].values[0]
    fig.add_annotation(
        x=median_value,
        y=position,
        text=f"{median_value:.0f}",
        showarrow=False,
        font=dict(color="black", size=12),
        opacity=0.5,
        align="center",
        xshift=10
    )

fig.show()

## Распределение желаемой зарплаты по вузам

Рассмотрим топ-30 самых встречающихся у соискателей ВУЗов

In [31]:
education_counts = preprocessed_df['Образование и ВУЗ'].value_counts()

filtered_education = education_counts[:30].index

filtered_df = preprocessed_df[preprocessed_df['Образование и ВУЗ'].isin(filtered_education)]

In [32]:
median_salary_by_uni = filtered_df.groupby('Образование и ВУЗ')['ЗП'].median().reset_index()
median_salary_by_uni.rename(columns={'ЗП': 'Медиана'}, inplace=True)
median_salary_by_uni.head()

Образование и ВУЗ   Медиана
0  Владимирский государственный университет имени...   80000.0
1    Казанский (Приволжский) федеральный университет   47500.0
2                                  Колледж связи №54  100000.0
3  Московский авиационный институт (Национальный ...   90000.0
4  Московский авиационный институт (национальный ...  130000.0

In [33]:
fig = px.box(
    filtered_df,
    y='Образование и ВУЗ',
    x='ЗП',
    title='Распределение зарплат соискателей по уровню образования',
    labels={
        'Образование и ВУЗ': 'Образование и ВУЗ',
        'ЗП': 'Заработная плата'
    },
    height=1000,
    width=1500,
    color="Образование и ВУЗ",
    orientation="h"
)

fig.update_layout(
    xaxis=dict(
        tickformat='.2s',
        title="Заработная плата",
    ),
    yaxis_title="ВУЗ",
    title_x=0.5,
    showlegend=False
)

for position in median_salary_by_uni['Образование и ВУЗ']:
    median_value = median_salary_by_uni.loc[median_salary_by_uni['Образование и ВУЗ'] == position, 'Медиана'].values[0]
    fig.add_annotation(
        x=median_value,
        y=position,
        text=f"{median_value:.0f}",
        showarrow=False,
        font=dict(color="black", size=12),
        opacity=0.5,
        align="center",
        xshift=10
    )

fig.show()

In [35]:
# preprocessed_df[preprocessed_df["Город"] == "Зеленоград"]

## Выводы



In [37]:
n = preprocessed_df[preprocessed_df["Искомая позиция"].isin(["Golang Developer",
                                                             "Java Developer",
                                                             "Python Developer"])].shape[0]
n / preprocessed_df.shape[0]

0.5719557195571956

In [38]:
n = preprocessed_df[preprocessed_df["Искомая позиция"].isin(["Machine Learning Engineer",
                                                             "Data Scientist",
                                                             "Data Engineer",
                                                             "Data Analyst"])].shape[0]
n / preprocessed_df.shape[0]

0.04164470216130733

In [39]:
8721 / (6455 + 8721)

0.5746573537163943

In [41]:
n = preprocessed_df[preprocessed_df["Город"].isin(["Москва",
                                                   "Санкт-Петербург"])].shape[0]
n / preprocessed_df.shape[0]

0.5005271481286241

Доля разработчиков на языках Go, Java и Python составляет более 57% рынка за 2024 год. Причем наибольшее количество резюме приходится на язык Go, что подтверждает его растущую популярность. Количество соискателей в области Big Data (MLE, DS, DE, DA) составляют заметно меньшую долю рынка (менее 5 %). Это может быть связано с тем, что на вакансии в этой области помимо знаний различных языков программирования и фреймворков требуется и понимание основ математического анализа, теории вероятностей и математической статистики, что делает эту область более сложной для освоения.

Большая части соискателей (более 57%) предпочитает скрывать размер желаемой зарабтной платы. Это может быть связано с тем, что многие соискатели желают получить наиболее выгодные условия работы по итогам собеседования.

Медианные зарпалты варьируются от 70 т. р. у Python разработчиков до 170 т. р. у DevOps-инженеров. Это может быть объяснено тем, что Python имеет достаточно простой синтаксис и может быть применен для самых различных задач, количество соискателей на эту позицию достаточно высокое. Причем многие из них не имеют боольшого опыта работы, что подтверждается наименьшим медианным возрастом Python-разработчиков в сравнении с другими вакансиями. Напротив, работа в качестве DevOps-инженера требует владения больишм количеством различных навыков, что повышает уровень медианной зарплаты специалистов.

Ожидаемо, более 50% резюме соискателей приходится на Санкт-Петербург и Москву. Также среди лидеров по количеству вакансий в области IT можно выделить крупные города, такие как Казань, Краснодар, Нижний Новгород, Новосибирск, Екатеринбург.\

Наибольшая медианная заработная плата (130 т. р.) среди представленных в данных городов соответствует Зеленограду. Это может быть связано с тем, что на территории Зеленограда находится особая экономическая зона, в которой зарегистрировано более 200 высокотехнологичных компаний. Наименьшие медианные заработные платы соответстсвуют небольшим городам, которые находятся недостаточно близко к крупным центрам (Ярославль, Рязань).

Самым часто встречающимся навыком среди соискателей является Git, так как навык работы с системами контроля версий необходимо всем IT-специалистам независимо от их специализации. В первую пятерку также входят SQL, Python, JavaScript и PostgreSQL. Картина самых распространенных навыков отличается в зависимости от искомой позиции. Так, для Frontend-разработчиков в пятерку входят навыки работы с JavaScript, Git, React, TypeScript и HTML, что ожидаемо, так как это самые распространенные инструменты разработки фронтенда. Для DevOps-инженеров в пятерке самых распространенных навыков вместо JavaScript присутствует Linux.

Медианный возраст соискателей для 6 позиций из 11 не превышает 30 лет. На основании этого факта можно сделать вывод о популярности IT-рынка среди молодежи.

Наибольшие значения ожидаемого уровня заработной платы соответствуют выпускникам технических ВУЗов, занимающих высокие позиции в рейтингах учебных заведений, таких как МИФИ, НИУ ВШЭ, СПБПУ, ТПУ, ЛЭТИ. Достаточно невысокий уровень ожидаемой заработной платы в МГУ и СПБГУ может быть объяснен тем, что многие выпускники этих вузов ищут работу на IT-рынке, хотя получили образование в другой сфере (математика, физика, химия и так далее).

По итогам исследования можно сделать вывод, что российский IT-рынок наиболее развит в Москве и Санкт-Петербурге, а также в других крупных городах. Подавляющее большинство соискателей ищут работу в качестве разработчиков на языках Go, Java и Python. Наибольшие медианные ожидаемые заработные платы приходятся на крупыне центры и города, расположенные в непосредственной близовсти от них. Среди наиболее распространенных навыков для всех специализаций встречается система контроля версий Git, другие навыки могут сильно варьироваться в зависимости от специализации. Медианный возраст соискателей по всем специализациям не превышает 40 лет, а по 6 из 11 специализациям не превышает 30 лет, что свидетельствует о том, что IT-рынок очень популярен у молодежи.